grin/src/Transformations/ExtendedSyntax/Conversion.hs

```diff
@@ -257,14 +262,27 @@ convertToNew :: Exp -> New.Exp
 convertToNew = convert . nameEverything
 
 nameEverything :: Exp -> Exp
-nameEverything = nodeArgumentNaming
-               . bindNormalisation
-               . appArgumentNaming
-               . bindNormalisation
-               . fst . bindingPatternSimplification
-               . bindNormalisation
-               . fst . producerNameIntroduction
-               . bindNormalisation
+nameEverything
+  = sdve
+  . copyPropagation
+  . bindNormalisation
+  . nodeArgumentNaming
+  . bindNormalisation
+  . appArgumentNaming
+  . bindNormalisation
+  . fst . bindingPatternSimplification
+  . bindNormalisation
+  . fst . producerNameIntroduction
+  . bindNormalisation
+
+  where
+    -- SDVE that infers the type env and the effect map
```

Comment:
Why do we need a Simple Dead Variable Elimination for the renaming? This looks suspicious :)

Reply:
See the comment in copy propagation. CP does not remove the resulting dead bindings. IMO, this should be fixed in the future.